neuralspot/ns-uart/src/apollo3/ns_uart.c

```diff
@@ -0,0 +1,78 @@
+#include "am_mcu_apollo.h"
+#include "am_bsp.h"
+#include "am_util.h"
+#include "ns_uart.h"
+
+void am_uart_isr(void)
+{
+    // Service the FIFOs as necessary, and clear the interrupts.
+    uint32_t ui32Status, ui32Idle;
+    am_hal_uart_interrupt_status_get(phUART, &ui32Status, true);
+    am_hal_uart_interrupt_clear(phUART, ui32Status);
+    am_hal_uart_interrupt_service(phUART, ui32Status, &ui32Idle);
+}
+
+uint32_t init_uart(am_hal_uart_config_t *uart_config)
+{
+    // Set the clock frequency.
+    am_hal_clkgen_control(AM_HAL_CLKGEN_CONTROL_SYSCLK_MAX, 0);
```

Comment:
This through line 25 doesn't belong here (ns_peripheral_power controls power, and this uart code will override that.)